ts/private/ts_lib.bzl

```diff
@@ -87,6 +87,18 @@ https://docs.aspect.build/rulesets/aspect_rules_js/docs/js_library#deps for more
         mandatory = True,
         allow_single_file = [".json"],
     ),
+    "isolated_typecheck": attr.bool(
+        doc = """\
+        Whether type-checking should be a separate action.
+
+        This allows the transpilation action to run without waiting for typings from dependencies.
+
+        Requires the typescript 5.6 [noCheck](https://www.typescriptlang.org/tsconfig#noCheck).
+
+        Requires tsconfig [isolatedDeclarations](https://www.typescriptlang.org/tsconfig#isolatedDeclarations)
```

Comment:
"Requires [isolatedDeclarations](https://www.typescriptlang.org/tsconfig#isolatedDeclarations) to be set in tsconfig to so that declarations are emitted without dependencies."

Reply:
Also mention that this flag can cause compile errors because of <insert reasons> and that it is best to front-load that change before flipping this to True